integr8ly/delorean
# DEL-224 Add release osd-addon command to automatically updating the managed-tenants repo

**What**

- Added the _osd-addon-release_ to automatically update the managed-tenants repo on release

**Steps to verify it**

1. Fork the manage-tenants repo

   https://gitlab.cee.redhat.com/service/managed-tenants

   ```
   export MANAGED_TENANTS_ORIGIN=myuser/managed-tenants
   export MANAGED_TENANTS_FORK=myuser/managed-tenants
   ```

2. Fork the integreatly-operator repo and create two fake releases (tags) or reuse my fork where I've created a fake 2.1.0-rc1 and 2.1.0 releases

   https://github.com/b1zzu/integreatly-operator/releases

   ```
   export INTEGREATLY_ORG=b1zzu
   ```

3. Obtain a GitLab personal token

   https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html

   ```
   export GITLAB_TOKEN=mypersonaltoken
   ```

4. Update the stage project for a build release (2.1.0-rc1)

   ```
   go run main.go release osd-addon \
       --org $INTEGREATLY_ORG \
       --gitlab-token $GITLAB_TOKEN \
       --version 2.1.0-rc1 \
       --managed-tenants-origin $MANAGED_TENANTS_ORIGIN \
       --managed-tenants-fork $MANAGED_TENANTS_FORK
   ```

5. Verify the created MR which will be printed on the console

   ex: https://gitlab.cee.redhat.com/dbizzarr/managed-tenants/merge_requests/1

6. Merge the MR in your fork

7. Update the stage, edge and stable projects for a final release (2.1.0)

   ```
   go run main.go release osd-addon \
       --org $INTEGREATLY_ORG \
       --gitlab-token $GITLAB_TOKEN \
       --version 2.1.0 \
       --managed-tenants-origin $MANAGED_TENANTS_ORIGIN \
       --managed-tenants-fork $MANAGED_TENANTS_FORK
   ```

8. Verify the created MRs which will be printed on the console

   Ex:

   - https://gitlab.cee.redhat.com/dbizzarr/managed-tenants/merge_requests/2
   - https://gitlab.cee.redhat.com/dbizzarr/managed-tenants/merge_requests/3
   - https://gitlab.cee.redhat.com/dbizzarr/managed-tenants/merge_requests/4
